### ISSUE TITLE
how to use my own loss to compute grad?

### ISSUE THREAD

**Issue by YangWun** [apache/incubator-mxnet#3917]
Now,  I want to use my own loss Class "MyLoss", but where to use it ? used like below?
model.fit(X=X_train, eval_metric=[MyLoss],
               batch_end_callback=mx.callback.Speedometer(batch_size,50),)

**Comment by piiswrong** [comment]
metric is only for reporting. You need to write a loss layer

**Comment by VoVAllen** [comment]
mx.sym.MakeLoss

**Comment by Andrjusha** [comment]
Hi guys! I'm also wondering how to write a loss layer. I've seen the example/numpy-ops/custom_softmax.py, but I don't understand where exactly is the loss function is applied? It looks like it just applies softmax, but after we need a loss function (e.g. log-loss), where is it defined? I'd like to change it.
And also I'd like that my loss layer works not just on CPU, but on multiple GPU, could you please tell me how to do it?

**Comment by whaozl** [comment]
yes, eval_metric is used to report the model train accuracy rate, eval_metric doesn't influence the model training. I also don't know how to define loss lay. I doesn't know what loss function on the Softmax lay.

**Comment by yxzf** [comment]
@piiswrong How to write a loss layer? I don't find an example about the loss layer.

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
